Makefile

```diff
@@ -18,6 +18,11 @@ coverage_r: build_r
 	echo "Done calculating coverage"
 	open coverage.html
 
+lint_r:
+	echo "Linting R source..."
+	TASK=rpkg .ci/lint.sh
```

Comment:
Can you come back and change this? This should be:

```make
lint_r:
    echo "Linting R source..."
    Rscript .ci/lint.R $$(pwd)
    echo "Done linting"
```

So that when someone runs `make lint_r`, the linting code will run. The `TASK` thing is a special environment variable that only works in the context of how we have set up our Continuous Integration builds using `.ci/test.sh`.

Reply:
I'll take this into account. 

I would be interested in fixing the linting checks. What do you think of using [styler](https://github.com/r-lib/styler), so that every developer could use the same rules ?

Reply:
ok great, thanks!

If you want to use `styler` as a way to address these checks, you are welcome to. But I do not want to make that project a part of this one's CI and I don't want to introduce any changes that are not directly related to passing the linters that you're proposing. So for example I don't want to see our comma-first lists changed to comma-last.

Reply:
You can use [precommit](https://lorenzwalthert.github.io/precommit/) so styler will run automatically before you make any commit. If you don't like it, just skip the hook with `--no-verify`(see link above for details).

Reply:
Thanks for the suggestion! Looks like an interesting project.

I would like to avoid using pre-commit hooks for this project, especially any that require installing additional software beyond simple shell scripts. This project has both Python and R components, and I do not want someone developing on the Python side to need an R environment locally, or vice versa.

I would like to minimize the barriers to entry for new contributors to the project, and I think that pushing checks into CI instead of making assumptions about contributors' local environments is one way to accomplish that goal.

Reply:
Sure. Then you can use https://github.com/r-lib/ghactions and https://github.com/r-lib/actions, it has a styler example if that's preferred.